src/components/Forum/ForumHeader.js

```diff
@@ -0,0 +1,199 @@
+import {
+  followForum,
+  getForumFollowData,
+  unfollowForum,
+  updateNotification
+} from "../../services/Forum/HandleForum";
+import { Dimensions } from "react-native";
+import { Icon } from "react-native-elements";
+import { useCallback, useEffect, useState } from "react";
+import Warning from "../../components/Forum/Warning";
+import styled from "styled-components/native";
+import { debounce } from "lodash";
+
+const ForumHeader = ({ id, img, title, banner, desc, isOwner }) => {
+  const [isFollowed, setIsFollowed] = useState(false);
+  const [isNotificationOn, setIsNotificationOn] = useState(false);
+  const debouncedUpdateNotification = useCallback(
+    debounce(updateNotification, 300),
+    []
+  );
+
+  useEffect(() => {
+    getForumFollowData(id, (data) => {
+      setIsFollowed(data.isFollowed);
+      setIsNotificationOn(data.isNotificationOn);
+    });
+  }, []);
+
+  function handleButtonClick() {
+    isFollowed
+      ? Warning(() =>
+          unfollowForum(id, () => {
+            setIsFollowed(!isFollowed);
+            setIsNotificationOn(false);
+          })
+        )
+      : followForum(id, () => setIsFollowed(!isFollowed));
+  }
+
+  function handleNotificationClick() {
```

Comment:
After editing notification permission, it would be better to show an alert to avoid ambiguity.